src/Core/Math/Rectangle.js

```diff
@@ -3,14 +3,61 @@
  * Class: Rectangle
  */
 
-function Rectangle() {
```

Comment:
rectangle is used in region check with only 4 coords: west, east, south, north. It's faster than bbox check. 
From rectangle we can have the bbox in 2D but in theory rectangle is not bbox. I refer to keep them separately to make code more clear.